build-ios-shared.sh报错，如何解决

mac下运行 `sh build-ios-shared.sh` 编译报错：
`
[100%] Linking CXX shared library ../../lib/libsherpa-onnx-c-api.dylib
ld: warning: duplicate -rpath '@loader_path' ignored
ld: warning: ignoring duplicate libraries: '-lm'
ld: library 'onnxruntime' not found
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
gmake[2]: *** [sherpa-onnx/c-api/CMakeFiles/sherpa-onnx-c-api.dir/build.make:107: lib/libsherpa-onnx-c-api.dylib] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:1311: sherpa-onnx/c-api/CMakeFiles/sherpa-onnx-c-api.dir/all] Error 2
gmake: *** [Makefile:156: all] Error 2
`

--- a/.github/workflows/build-xcframework.yaml
+++ b/.github/workflows/build-xcframework.yaml
@@ -43,6 +43,12 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
+      - name: Build iOS shared
+        if: matrix.with_tts == 'ON'
+        shell: bash
+        run: |
+          ./build-ios-shared.sh
+
       - name: Build iOS
         if: matrix.with_tts == 'ON'
         shell: bash

--- a/.github/workflows/build-xcframework.yaml
+++ b/.github/workflows/build-xcframework.yaml
@@ -47,6 +47,7 @@ jobs:
         if: matrix.with_tts == 'ON'
         shell: bash
         run: |
+          export CMAKE_VERBOSE_MAKEFILE=ON
           ./build-ios-shared.sh
 
       - name: Build iOS

--- a/build-ios-shared.sh
+++ b/build-ios-shared.sh
@@ -50,7 +50,7 @@ if [[ ! -f build/simulator_x86_64/install/lib/libsherpa-onnx-c-api.dylib ]]; the
     -DBUILD_PIPER_PHONMIZE_TESTS=OFF \
     -DBUILD_ESPEAK_NG_EXE=OFF \
     -DBUILD_ESPEAK_NG_TESTS=OFF \
-    -S .. \
+    -S .. -D CMAKE_VERBOSE_MAKEFILE=ON \
     -DCMAKE_TOOLCHAIN_FILE=./toolchains/ios.toolchain.cmake \
     -DPLATFORM=SIMULATOR64 \
     -DENABLE_BITCODE=0 \
@@ -81,7 +81,7 @@ if [[ ! -f build/simulator_arm64/install/lib/libsherpa-onnx-c-api.dylib ]]; then
     -DBUILD_PIPER_PHONMIZE_TESTS=OFF \
     -DBUILD_ESPEAK_NG_EXE=OFF \
     -DBUILD_ESPEAK_NG_TESTS=OFF \
-    -S .. \
+    -S .. -D CMAKE_VERBOSE_MAKEFILE=ON \
     -DCMAKE_TOOLCHAIN_FILE=./toolchains/ios.toolchain.cmake \
     -DPLATFORM=SIMULATORARM64 \
     -DENABLE_BITCODE=0 \
@@ -114,7 +114,7 @@ if [[ ! -f build/os64/install/lib/libsherpa-onnx-c-api.dylib ]]; then
     -DBUILD_PIPER_PHONMIZE_TESTS=OFF \
     -DBUILD_ESPEAK_NG_EXE=OFF \
     -DBUILD_ESPEAK_NG_TESTS=OFF \
-    -S .. \
+    -S .. -D CMAKE_VERBOSE_MAKEFILE=ON \
     -DCMAKE_TOOLCHAIN_FILE=./toolchains/ios.toolchain.cmake \
     -DPLATFORM=OS64 \
     -DENABLE_BITCODE=0 \

--- a/build-ios-shared.sh
+++ b/build-ios-shared.sh
@@ -19,6 +19,10 @@ if [ "$SHERPA_ONNX_GITHUB_MIRROW" == true ]; then
     SHERPA_ONNX_GITHUB=hub.nuaa.cf
 fi
 
+if [ ! -z CMAKE_VERBOSE_MAKEFILE ]; then
+  CMAKE_VERBOSE_MAKEFILE=ON
+fi
+
 if [ ! -f $onnxruntime_dir/onnxruntime.xcframework/ios-arm64/onnxruntime.a ]; then
   mkdir -p $onnxruntime_dir
   pushd $onnxruntime_dir
@@ -50,7 +54,7 @@ if [[ ! -f build/simulator_x86_64/install/lib/libsherpa-onnx-c-api.dylib ]]; the
     -DBUILD_PIPER_PHONMIZE_TESTS=OFF \
     -DBUILD_ESPEAK_NG_EXE=OFF \
     -DBUILD_ESPEAK_NG_TESTS=OFF \
-    -S .. -D CMAKE_VERBOSE_MAKEFILE=ON \
+    -S .. -D CMAKE_VERBOSE_MAKEFILE=$CMAKE_VERBOSE_MAKEFILE \
     -DCMAKE_TOOLCHAIN_FILE=./toolchains/ios.toolchain.cmake \
     -DPLATFORM=SIMULATOR64 \
     -DENABLE_BITCODE=0 \
@@ -81,7 +85,7 @@ if [[ ! -f build/simulator_arm64/install/lib/libsherpa-onnx-c-api.dylib ]]; then
     -DBUILD_PIPER_PHONMIZE_TESTS=OFF \
     -DBUILD_ESPEAK_NG_EXE=OFF \
     -DBUILD_ESPEAK_NG_TESTS=OFF \
-    -S .. -D CMAKE_VERBOSE_MAKEFILE=ON \
+    -S .. -D CMAKE_VERBOSE_MAKEFILE=$CMAKE_VERBOSE_MAKEFILE \
     -DCMAKE_TOOLCHAIN_FILE=./toolchains/ios.toolchain.cmake \
     -DPLATFORM=SIMULATORARM64 \
     -DENABLE_BITCODE=0 \
@@ -114,7 +118,7 @@ if [[ ! -f build/os64/install/lib/libsherpa-onnx-c-api.dylib ]]; then
     -DBUILD_PIPER_PHONMIZE_TESTS=OFF \
     -DBUILD_ESPEAK_NG_EXE=OFF \
     -DBUILD_ESPEAK_NG_TESTS=OFF \
-    -S .. -D CMAKE_VERBOSE_MAKEFILE=ON \
+    -S .. -D CMAKE_VERBOSE_MAKEFILE=$CMAKE_VERBOSE_MAKEFILE \
     -DCMAKE_TOOLCHAIN_FILE=./toolchains/ios.toolchain.cmake \
     -DPLATFORM=OS64 \
     -DENABLE_BITCODE=0 \

--- a/sherpa-onnx/csrc/CMakeLists.txt
+++ b/sherpa-onnx/csrc/CMakeLists.txt
@@ -214,7 +214,7 @@ if(SHERPA_ONNX_ENABLE_GPU)
   )
 endif()
 
-if(BUILD_SHARED_LIBS)
+if(BUILD_SHARED_LIBS AND NOT DEFINED onnxruntime_lib_files)
   target_link_libraries(sherpa-onnx-core onnxruntime)
 else()
   target_link_libraries(sherpa-onnx-core ${onnxruntime_lib_files})

